File: lab03/lab03_task01/src/lab03_task01_req_node.cpp
#include "ros/ros.h"
//TODO: Include the header file rotate_point service (look in the package example_msg_srv)

int main(int argc, char **argv) {

    ros::init(argc, argv, "rotate_point_client");

    ros::NodeHandle nh;

    //TODO: Initialise the client and makes sure that it corresponds to the correct service

    //TODO: Initialise the service message

    while (nh.ok())
    {

        //TODO: Generate a random point
        //TODO: Generate a random quaternion (make sure its norm is 1)

        //TODO: Send a request.
	//TODO: Show the computational time between requests (from sending a request and getting a response).
        //TODO: Change the output value
        ROS_INFO("The input point (x, y, z) = (%.2f, %.2f, %.2f)\n", 0.0, 0.0, 0.0);
        ROS_INFO("The output point (x, y, z) = (%.2f, %.2f, %.2f)\n", 0.0, 0.0, 0.0);

        usleep(10000);
    }

    return 1;
}

File: lab03/lab03_task01/scripts/lab03_task01_req_node.py
#!/usr/bin/env python

import rospy
import random
##TODO: Include the header file rotate_point service (look in the package example_msg_srv)


def rotate_point_client():

    ##TODO: Wait for rotate_point service (look in the package example_msg_srv)

    while not rospy.is_shutdown():
        ##TODO: Initialise your client
        ##TODO: Generate a random point and a random quaternion (make sure its norm is 1)

        ##TODO: Compute the computational time between requests (from sending a request and getting a response).

        print 'The input point'
        ##TODO: Print out the input and the output points
        print 'The output point'


if __name__ == "__main__":
    try:
        rotate_point_client()
    except rospy.ROSInterruptException:
        pass
